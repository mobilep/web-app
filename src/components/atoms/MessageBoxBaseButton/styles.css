.MessageBoxBaseButton,
.MessageBoxBaseButton:hover {
	display: flex;
	justify-content: center;
	align-items: center;
	width: 50px;
	padding: 0;
	border: none;
	border-radius: 0;
	background-color: var(--white);
	color: var(--grey3);
}

.MessageBoxBaseButton svg {
	width: 30px;
}
