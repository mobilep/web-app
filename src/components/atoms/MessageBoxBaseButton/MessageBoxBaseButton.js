import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Button } from '..';
import './styles.css';

const getClassName = (className) => classNames('MessageBoxBaseButton', { [className]: className });

const MessageBoxBaseButton = ({ children, className, ...rest }) => {

	return (
		<Button { ...rest } className={getClassName(className)}>
			{children}
		</Button>
	);
};

MessageBoxBaseButton.propTypes = {
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.element]),
	className: PropTypes.string,
};

export default MessageBoxBaseButton;
