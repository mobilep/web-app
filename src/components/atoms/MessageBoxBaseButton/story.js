import React from 'react';
import { storiesOf } from '@storybook/react';
import { InboxIcon } from '../../../icons';

import MessageBoxBaseButton from './MessageBoxBaseButton';

storiesOf('1. Atoms / MessageBoxBaseButton', module)
	.add('MessageBoxBaseButton', () => (
		<MessageBoxBaseButton>
			<InboxIcon />
		</MessageBoxBaseButton>
	));
