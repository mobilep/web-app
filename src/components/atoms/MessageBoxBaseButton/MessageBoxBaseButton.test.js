import React from 'react';
import { shallow } from 'enzyme';

import MessageBoxBaseButton from './MessageBoxBaseButton';

describe('Atom - MessageBoxBaseButton', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<MessageBoxBaseButton>
				Text
			</MessageBoxBaseButton>
		);
		expect(wrapper).toMatchSnapshot();
	});

});

