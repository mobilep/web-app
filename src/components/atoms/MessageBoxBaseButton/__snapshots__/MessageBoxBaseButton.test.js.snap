// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - MessageBoxBaseButton Should render 1`] = `
<Button
  className="MessageBoxBaseButton"
>
  Text
</Button>
`;
